File: scripts/parse-svg.js
const fs = require('fs');
const path = require('path');
const { JSDOM } = require('jsdom');

async function processSVGs() {
  const dir = path.join(__dirname, '../fromfigma');
  const files = fs.readdirSync(dir).filter(f => f.match(/^\d+\.svg$/));

  const templates = [];

  for (const file of files) {
    const filePath = path.join(dir, file);
    const content = fs.readFileSync(filePath, 'utf-8');
    
    const dom = new JSDOM(content);
    const document = dom.window.document;
    
    const rects = Array.from(document.querySelectorAll('rect'));
    const mockups = [];
    
    for (const rect of rects) {
      const w = parseFloat(rect.getAttribute('width') || '0');
      const h = parseFloat(rect.getAttribute('height') || '0');
      
      // Identify mockups by aspect ratio and size
      // Typical iPhone is ~ 920x2000 (aspect ~ 0.46)
      // The shadow/frame is ~ 1030x2100 (aspect ~ 0.49)
      if (w > 800 && w < 1200 && h > 1800 && h < 2400) {
        // This is a mockup or its shadow
        const x = parseFloat(rect.getAttribute('x') || '0');
        const y = parseFloat(rect.getAttribute('y') || '0');
        const transform = rect.getAttribute('transform');
        
        mockups.push({ x, y, w, h, transform });
      }
    }
    
    templates.push({
      file,
      mockupsCount: mockups.length,
      mockups
    });
  }
  
  console.log(JSON.stringify(templates.slice(0, 2), null, 2));
}

processSVGs();

File: scripts/parse-svg.py
import xml.etree.ElementTree as ET
import os
import json
import sys

def parse_svgs():
    dir_path = 'fromfigma'
    files = [f for f in os.listdir(dir_path) if f.endswith('.svg') and f[:-4].isdigit()]
    
    templates = []
    
    ns = {'svg': 'http://www.w3.org/2000/svg'}
    
    for file in sorted(files, key=lambda x: int(x[:-4])):
        filepath = os.path.join(dir_path, file)
        tree = ET.parse(filepath)
        root = tree.getroot()
        
        mockups = []
        rects = root.findall('.//svg:rect', ns)
        
        for rect in rects:
            w_str = rect.get('width')
            h_str = rect.get('height')
            if w_str and h_str:
                w = float(w_str)
                h = float(h_str)
                
                # Check for mockups (around 920x2000)
                if 800 < w < 1200 and 1800 < h < 2400:
                    x = float(rect.get('x', '0'))
                    y = float(rect.get('y', '0'))
                    transform = rect.get('transform')
                    fill = rect.get('fill')
                    
                    mockups.append({
                        'x': x,
                        'y': y,
                        'w': w,
                        'h': h,
                        'transform': transform,
                        'fill': fill
                    })
        
        templates.append({
            'file': file,
            'mockups': mockups
        })
        
    print(json.dumps(templates[:2], indent=2))

if __name__ == "__main__":
    parse_svgs()
